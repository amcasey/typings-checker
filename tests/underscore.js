// This checks a few properties of the lodash typings.
// $ExpectType number
_.find([1, 2, 3], function (x) { return x * 1 == 3; });
// $ExpectError Operator '==' cannot be applied to types 'number' and 'string'.
_.find([1, 2, 3], function (x) { return x == 'a'; });
// $ExpectType number
_.find([1, 2, 3], 1);
// $ExpectError Property 'y' does not exist on type '{ x: number; }'.
_.find([{ x: 1 }, { x: 2 }, { x: 3 }], function (v) { return v.y == 3; });
// $ExpectType { x: number; }
_.find([{ x: 1 }, { x: 2 }, { x: 3 }], function (v) { return v.x == 3; });
