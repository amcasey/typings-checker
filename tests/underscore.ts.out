tests/underscore.ts: 5 / 5 checks passed.
